File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.js
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = require("fs").readFileSync(filePath).toString().split("\n");

const N = input.shift();

const mySet = new Set();

for (let i = 0; i < N; i++) {
  let person = input[i].split(" ");
  if (person[1] === "enter") mySet.add(person[0]);
  if (person[1] === "leave") mySet.delete(person[0]);
}

const answer = [...mySet].sort().reverse().join("\n");

console.log(answer);

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
N = int(input());

myList = set();

for i in range(N):
    name, state = input().split();
    if(state == 'enter') : myList.add(name);
    elif( state == 'leave') : myList.remove(name);
    
print("\n".join(sorted(myList,reverse=True)))